libs/sdk-ui-kit/src/Dialog/ShareDialog/test/testHelpers.tsx

```diff
@@ -0,0 +1,157 @@
+// (C) 2019 GoodData Corporation
+
+import React from "react";
+import { mount, ReactWrapper } from "enzyme";
+import { act } from "react-dom/test-utils";
+import { noop } from "lodash";
+import { recordedBackend, RecordedBackendConfig } from "@gooddata/sdk-backend-mockingbird";
+import { ReferenceRecordings } from "@gooddata/reference-workspace";
+import { AccessGranteeDetail, IWorkspaceUser, IWorkspaceUserGroup } from "@gooddata/sdk-backend-spi";
+import { IShareDialogLabels, IShareDialogProps, ISharedObject } from "../types";
+import { uriRef } from "@gooddata/sdk-model";
+import { ShareDialog } from "../ShareDialog";
+import { groupAll } from "../ShareDialogBase/test/GranteeMock";
+import { getGranteeItemTestId } from "../ShareDialogBase/utils";
+import { mapWorkspaceUserToGrantee } from "../shareDialogMappers";
+
+export const labels: IShareDialogLabels = {
+    accessTypeLabel: "lockControl label",
+    accessRegimeLabel: "underLenientControl label",
+    removeAccessGranteeTooltip: "removeAccessGranteeToolTip",
+    removeAccessCreatorTooltip: "removeAccessGranteeToolTip",
+};
+
+export const defaultSharedObject: ISharedObject = {
+    ref: uriRef("objRef"),
+    shareStatus: "private",
+    createdBy: undefined,
+};
+
+export const defaultProps: IShareDialogProps = {
+    backend: recordedBackend(ReferenceRecordings.Recordings),
+    workspace: "foo",
+    sharedObject: defaultSharedObject,
+    currentUserRef: uriRef("userRef"),
+    labels,
+    isLockingSupported: true,
+    locale: "en-US",
+    onApply: noop,
+    onCancel: noop,
+    onError: noop,
+};
+
+export const createComponent = async (
+    customProps: Partial<IShareDialogProps> = {},
+    users: IWorkspaceUser[] = [],
+    groups: IWorkspaceUserGroup[] = [],
+    grantees: AccessGranteeDetail[] = [],
+): Promise<ReactWrapper> => {
+    const config: RecordedBackendConfig = {
+        userManagement: {
+            users: {
+                users: users,
+            },
+            userGroup: {
+                userGroups: groups,
+            },
+            accessControl: {
+                accessList: grantees,
+            },
+        },
+    };
+
+    const backend = recordedBackend(ReferenceRecordings.Recordings, config);
+
+    const props: IShareDialogProps = { ...defaultProps, ...customProps, backend };
+
+    const wrapper = mount(<ShareDialog {...props} />);
+
+    await waitForComponentToPaint(wrapper);
+
+    return wrapper;
+};
+
+export function isDialogVisible(wrapper: ReactWrapper): boolean {
```

Comment:
Just a note for the next time: I find it better to wrap all these methods into a class and put wrapper as its constructor parameter. Ideally each helper can return itself (this) and you can chain these utility methods. It makes for much more readable tests.